tests/job/test_genericJob.py

```diff
@@ -525,6 +525,10 @@ def test_job_executor_cancel(self):
         j.run()
         j.server.future.cancel()
         j.refresh_job_status()
+        self.assertEqual(j.project.db.get_job_status(j.job_id), "submitted")
+        sleep(1)
+        self.assertEqual(j.project.db.get_job_status(j.job_id), "aborted")
+        self.assertEqual(j.status, "aborted")
```

Comment:
Somehow the status remains in `submitted` rather than being switched to `aborted`.